### DISCUSSION TITLE
Report - (4th Sem.) Week 7📢

### DISCUSSION THREAD

**Discussion by Nahlam4** [SE-TINF22B6/DHBWhub#749] [Weekly Report]
Greetings and welcome to our fifteenth progress report! We're thrilled to update you on the latest news and developments since last week! 🥳

### Clean Code Report 💧 

In this week, we summarized (and applied) our Clean Code conventions with the following principles:

#### 1. Following naming conventions

  We ended up with these naming conventions:
- Functions, classes and variables are written in 'CamelCase'.
- Packages only contain small letters and are divided by hyphens instead of whitespace.
- Links are also divided by hyphens instead of whitespace.

Here are some examples: 
![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/8d68c248-c4b8-4afa-88fc-bca0b687d1b9)

#### 2. Single Responsibility Principle

The SRP is quite simple. Each function and method is responsible for one particular action or Set of action. If requirements are changing, then only the respective method has to change, nothing more. We are trying to implement SRP by outsourcing commonly used logic as new methods. One example for this are the MapperClasses. Instead of creating the needed Objects within the methods, we use static methods of the respective MapperClass, which takes the source data as input and returns the needed data model. You can see this in the following images. Moreover, you can also see that the logic for getting the amount of comments as well as tags of each post is located in a seperate function:
![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/1e0accd1-5804-4b8c-8bb8-d7a7e86d4a18)
![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/ed7137da-f171-4996-b893-eec6bad7d2b9)

#### 3. Consistent Formatting and Indentation

It goes without saying that the code formatting should stay consistent, even after creating new features. This is also accompanied by the boy scout ethos 'Leaving the campground cleaner than you found it'. If a developer in our team forgot the correct formatting, the developer who comes across the mistake should either fix it or point it out.

#### 4. Principle of Least Astonishement

Methods or requests as well as their results shouldn't surprise our users. That means that we have to name links as well as methods self explanatory and if that is not possible, we need to lay out comments which explain the logic behind the corresponding function. In the following examples, it is quite trivial what kind of data each endpoint returns:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/86d5b516-48e2-47bf-9f77-daee5db6d6ee)

### Frontend Developments

We have now successfully implemented the display of the comments of a Post: 
![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/d02f9dc7-2dab-48f0-b797-3adbbc8d859b)

Also, we have implemented the Pop-up for creating a Post, which looks like this: 

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/06e9d978-b0c5-47ba-b217-bb1f80328b16)

This Popup only shows, when a User is logged in, otherwise the guest will get this message: 
![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/195f823a-9dec-44f0-86f1-8cfa108e636d)

As we are unsure about wether we should have written our blog until Tuesday rather than Thursday this week, we will come back to this Report and update you on the further Progress we made this Sprint. :trollface:

### Sprint 9: What have we achieved ❓ What's to come❔
✅ Frontend: Setting up Google OAuth Authorization buttons
✅ Frontend: Post component retrieves post data from backend
✅ Frontend: Adapt event component to retrieve data from backend
✅ Frontend: Adapt components to retrieve profile picture from backend
✅ Frontend: Implement tests for essential homepage logic
✅ Postman: Updated Postman Collection with new Request Methods
✅ Backend: Implemented concept for logging like activities
✅ Backend: Implemented notifications concept
✅ Backend: Retrieve most popular tags
✅ Backend: Adjusted SavedPost Endpoints
✅ Frontend: Implement Imprint page
✅ Frontend: Implement 'Create Post' popup window
✅ Frontend: Implement comment-component
✅ Frontend: Implement Terms of Service page
✅ Backend: Setup HTTPS for backend server
✅ Backend: Implement Endpoint for posts of a single user
🚧 Frontend: Adapting Sign-Up dialog
🚧 Frontend: Implement profile page
🚧 Frontend: Implement notifications window
🚧 Frontend: Implement Search Page
🚧 Frontend: Implement FAQ-page
🚧 Frontend: Implement friends page
🚧 Frontend: Implement privacy policy page
🚧 Frontend: Implement calendar page
🚧 Frontend: Enable registration via google
🚧 Frontend: Implement 'Forget password dialog'
🚧 Backend: Implement report function
🚧 Backend: Refine Post Endpoints
🚧 Backend: Insert all Course options
🚧 Remove Herobrine

**Comment by mirak55** [comment]
Good work on the your Report about Clean Code, you have briefly explained all your principles, and provided good documentation on how you implemented it in your own code. Keep up the good work, and looking forward to see new progress.

**Comment by Umbranecis** [comment]
Very nice work on the Code Cleanup and the Clean Code report. I also like your pop up for the post, it looks very nice to use
